File: src/cleansia_ios/CleansiaCustomer/Sources/Features/Disputes/DisputeFormConstants.swift
import Foundation

enum DisputeFormConstants {
    static let descriptionMinLength = 10
    static let descriptionMaxLength = 2000
    static let messageMaxLength = 2000

    /// Mirrors the backend `UploadDisputeEvidenceCommand` validator exactly: the
    /// 10 MiB cap is checked AFTER compression, the type set is the accepted
    /// MIME whitelist. Both are FAIL-CLOSED on the client (Gate-SEC R11) — a
    /// doomed request is rejected before the network with a localized error.
    static let maxEvidenceBytes = 10 * 1024 * 1024
    static let allowedEvidenceContentTypes: Set<String> = [
        "image/jpeg",
        "image/png",
        "image/webp",
        "application/pdf"
    ]
}

File: src/cleansia_android/customer-app/src/main/java/cz/cleansia/customer/features/disputes/DisputeFormConstants.kt
package cz.cleansia.customer.features.disputes

object DisputeFormConstants {
    const val DESCRIPTION_MIN_LENGTH = 10
    const val DESCRIPTION_MAX_LENGTH = 2000

    /**
     * Mirrors the backend evidence validator, so a doomed upload never reaches the network — and if
     * the whitelist grows there it must grow here. -> /flows/cancellation-refund-dispute
     */
    const val EVIDENCE_MAX_BYTES: Int = 10 * 1024 * 1024
    val EVIDENCE_ALLOWED_MIME_TYPES: Set<String> = setOf(
        "image/jpeg",
        "image/jpg",
        "image/png",
        "image/webp",
        "application/pdf",
    )
}
